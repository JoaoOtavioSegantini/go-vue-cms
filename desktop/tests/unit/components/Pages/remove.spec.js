import { mount } from "@vue/test-utils";
import RemovePage from "@/components/Pages/Remove.vue";
import { createStore } from "vuex";

describe("RemovePage.vue", () => {
  afterEach(() => {
    jest.restoreAllMocks();
  });

  window.axios = {
    get: jest
      .fn(() => {
        return {
          then: jest.fn(() => "your faked response"),
        };
      })
      .mockName("axiosget"),
    delete: jest
      .fn(() => {
        return {
          then: jest.fn(() => "your faked response"),
        };
      })
      .mockName("axiosPut"),
  };

  const mockRoute = {
    params: {
      id: 1,
    },
  };

  const mockRouter = {
    push: jest.fn(),
  };

  const createVuexStore = () => {
    return createStore({
      state() {
        return {
          Pages: {
            onePage: {
              ID: "1",
              title: "Post 1",
              body: "A simple page text body",
              slug: "simple-slug-text",
            },
          },
        };
      },
      mutations: {
        updatePage(state, res) {
          state.onePage = res;
        },
      },
      actions: {
        getPage(ctx, res) {
          ctx.commit("updatePage", res);
        },
        removePage(ctx, res) {},
      },
    });
  };

  function factory() {
    const store = createVuexStore();

    return mount(RemovePage, {
      global: {
        plugins: [store],
        mocks: {
          axios: window.axios,
          $route: mockRoute,
          $router: mockRouter,
        },
      },
    });
  }

  it("renders props.msg when passed", async () => {
    const msg =
      "AÇÕES Post 1 Remoção de página do site Tem certeza que quer remover esta página, essa ação não poderá ser desfeita! Não removerApagar definitivamente";
    const wrapper = factory();

    await wrapper.vm.remove();

    expect(wrapper.text()).toMatch(msg);

    await wrapper.findAll("a")[1].trigger("click");
    await wrapper.findAll("a")[0].trigger("click");
    console.log(wrapper.emitted());

    expect(wrapper.html()).toMatchSnapshot();
  });
});
