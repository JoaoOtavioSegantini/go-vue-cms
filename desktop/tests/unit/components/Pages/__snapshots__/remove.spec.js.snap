// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemovePage.vue renders props.msg when passed 1`] = `
"<div class=\\"container-fluid\\">
  <div class=\\"row\\">
    <div class=\\"col-1 nav\\">
      <p class=\\"display-6\\"><small>AÇÕES</small></p><a href=\\"/pages/1\\" class=\\"btn btn-secondary\\" title=\\"Voltar\\"><i class=\\"fa fa-arrow-circle-left\\"></i></a>
    </div>
    <div class=\\"col-11\\">
      <div class=\\"card\\">
        <div class=\\"card-body\\">
          <h5 class=\\"title\\"><i class=\\"fa fa-file-text-o\\"></i> Post 1 <small>Remoção de página do site</small></h5>
          <div class=\\"alert alert-danger\\"> Tem certeza que quer remover esta página, essa ação não poderá ser desfeita! </div>
        </div>
        <div class=\\"card-footer\\"><a href=\\"/pages/1\\" class=\\"btn btn-secondary\\">Não remover</a><a href=\\"\\" class=\\"btn btn-primary\\">Apagar definitivamente</a></div>
      </div>
    </div>
  </div>
</div>"
`;
